src/libs/highlevel/README.md

```diff
@@ -0,0 +1,154 @@
+
+# High-Level API
+
+## Introduction
+
+The goal of the High-Level API is to increase the usability of libelektra for developers who want to integrate Elektra into their applications. Projects usually do not want to use low-level APIs. `KDB` and `KeySet` are useful for plugins and to implement APIs but not to be directly used in applications. The High-Level API should be extremely easy to get started with and at the same time it should be hard to use it in a wrong way. This tutorial gives an introduction for developers who want to elektrify their application using the High-Level API.
+
+## Setup
+
+First you have to add `elektra-highlevel` to the linked libraries of your application. To be able to use it in your source file, just include the main header with `#include "elektra.h"` at the top of your file.
+
+## Core Concepts
+
+### Struct `Elektra`
+`Elektra` is the handle you use to access the underlying KDB. All key-value read and write operations expect this handle to be passed as in as a parameter. To create the handle, you can simply write:
+
+```c
+ElektraError * error = NULL;
+Elektra * elektra = elektraOpen ("/sw/yourApplication", &error);
+```
+
+Please replace `"/sw/yourApplication"` with an appropriate value for your application (see [Namespaces](libelektra/doc/tutorials/namespaces.md) for more information).
+
+The passed in `ElektraError` can be used to check for initialization errors. You can detect initialization errors by comparing it to NULL after calling `elektraOpen`:
+
+```c
+if (error != NULL) 
+{
+  // An error occured
+}
+```
+
+In order to give Elektra the chance to clean up its all allocated ressources, you have to close your instance, when you are done using it, by calling:
+
+```c
+elektraClose (elektra);
+```
+
+Notice that Elektra is not thread-safe. If you have mutliple threads accessing key-values, create a separate handle for each thread to avoid concurrency issues.
+
+### Struct `ElektraError`
+The library is designed to hide as many problems a developer can encounter when usign KDB directly. However it is not possible to hide all those issues. As with every library, things can go wrong and there needs to be a way react to errors once they have occurred at runtime. Therefore the High-Level API introduces a simplified struct called `ElektraError`, which encapsulates all information neccessary for the developer to handle runtime-errors appropriately in the application.
+
+Functions that can produce errors accept an ElektraError pointer as parameter, for example:
+
+```c
+Elektra * elektraOpen (const char * application, ElektraError ** error);
+```
+
+In most cases you'll want to set the error variable to NULL before passing it to the function. You can do this either by declaring and initializing a new variable with `ElektraError * error = NULL` or by reusing an already existing error variable by resetting it with `elektraErrorFree (error)`. 
```

Comment:
Thank you for pointing this out! In fact it was necessary. I changed the name to elektraErrorReset and the function now in addition resets the passed error variable to NULL. Seems to me more intuitive this way.